completion/sdd

```diff
@@ -0,0 +1,87 @@
+# sdd completion
+#
+# @author: Sergei Eremenko (https://github.com/SmartFinn)
+# @license: MIT license (MIT)
+# @link: https://github.com/pylipp/sdd
+
+_sdd_list_available() {
+    local app_name
+
+    command -v sdd >/dev/null || return 1
+
+    while read -r _ app_name; do
+        [ -n "$app_name" ] || continue
+        printf '%s\n' "$app_name"
+    done < <(sdd list --available)
+
+    return 0
+}
+
+_sdd_list_installed() {
+    local app_name
+
+    command -v sdd >/dev/null || return 1
+
+    while IFS='=' read -r app_name _; do
+        [ -n "$app_name" ] || continue
+        printf '%s\n' "$app_name"
+    done < <(sdd list --installed)
+
+    return 0
+}
+
+_sdd_completion() {
+    local cur prev word cword
+    local -a words=()
+    local -a opts=(
+        install
+        list
+        uninstall
+        upgrade
+        -h --help
+        -V --version
+    )
+    local -a list_opts=(
+        -a --available
+        -i --installed
+        -u --upgradable
+    )
+
+    _init_completion || return
+
+    for word in "${words[@]}"; do
+        case "$word" in
+        install)
+            COMPREPLY=( $(compgen -W "$(_sdd_list_available)" -- "$cur") )
+            return 0
+            ;;
+        uninstall|upgrade)
+            COMPREPLY=( $(compgen -W "$(_sdd_list_installed)" -- "$cur") )
```

Comment:
Using the output of `sdd list -u` as completion options for `sdd upgrade` is probably delayed due to downloading data, and hence it might appear as if the program hangs. What do you think?

Reply:
Yeah, on average it takes 2 sec. × N apps. Not hard to calculate that 2 sec. × 19 apps ~ 38 seconds delay. It can be easily improved to 2-3 seconds with parallel downloading. I already implemented it in my fork. It may be I'll add it here later.